File: common/src/main/java/com/changanford/common/chat/utils/LogUtil.java
package com.changanford.common.chat.utils;

import android.util.Log;

import com.changanford.common.BuildConfig;

public   class LogUtil {

    public static void d(String msg) {
        Log.d("chatui", msg);
    }
    public static void d(String tag, String msg) {
        if (BuildConfig.DEBUG)
            Log.d(tag, msg);
    }
}

File: common/src/main/java/com/changanford/common/util/LogUtil.kt
package com.changanford.common.util

import android.util.Log

/**
 * @Author: lcw
 * @Date: 2020/8/10
 * @Des:
 */
/**
 * 截断输出日志
 * @param
 */
fun longLog(tag: String?, mMsg: String?) {
    var msg = mMsg
    if (tag == null || tag.isEmpty() || msg == null || msg.isEmpty()
    ) return
    val segmentSize = 3 * 1024
    val length = msg.length.toLong()
    // 长度小于等于限制直接打印
    if (length <= segmentSize) {
        Log.d(tag, msg)
    } else {
        // 循环分段打印日志
        while (msg!!.length > segmentSize) {
            val logContent = msg.substring(0, segmentSize)
            msg = msg.replace(logContent, "")
            Log.d(tag, logContent)
        }
        // 打印剩余日志
        Log.d(tag, msg)
    }
}